File: app/src/main/java/me/sargunvohra/android/diningcourts/main/MainApp.java
package me.sargunvohra.android.diningcourts.main;

import android.app.Application;

import me.sargunvohra.android.diningcourts.module.ApiModule;
import me.sargunvohra.android.diningcourts.module.AppModule;

public class MainApp extends Application {

    private MainComponent mainComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        mainComponent = DaggerMainComponent.builder()
                .appModule(new AppModule(this))
                .apiModule(new ApiModule("http://api.hfs.purdue.edu/menus/v2/"))
                .build();
    }
}

File: app/src/main/java/me/sargunvohra/android/diningcourts/main/MainApp.kt
package me.sargunvohra.android.diningcourts.main

import com.karumi.rosie.application.RosieApplication
import me.sargunvohra.android.diningcourts.config.dagger.AppModule
import nl.komponents.kovenant.android.androidUiDispatcher
import nl.komponents.kovenant.ui.KovenantUi

class MainApp : RosieApplication() {
    override fun getApplicationModules() = listOf(AppModule())

    override fun onCreate() {
        super.onCreate()
        KovenantUi.uiContext {
            dispatcher = androidUiDispatcher()
        }
    }
}